File: src/SkyBlock/enchants/sword/Thunderbolt.php
<?php


namespace SkyBlock\enchants\sword;


use pocketmine\entity\effect\EffectInstance;
use pocketmine\entity\effect\VanillaEffects;
use pocketmine\event\entity\EntityDamageEvent;
use pocketmine\player\Player;
use pocketmine\Server;

class Thunderbolt extends BaseMeleeEnchant {

    static int $id = 179;

    public function isApplicableTo(Player $holder) : bool {
        return mt_rand(1, 70) === 1;
    }

    public function getLevel(int $level = 1) : int {
        return (($int = ceil($level / 2)) > 7) ? 8 : $int;
    }


    public function onActivation(Player $player, Player $attacker, EntityDamageEvent $ev, int $enchantmentlevel) : void {
        $attacker->getEffects()->remove(VanillaEffects::REGENERATION());
        $attacker->getEffects()->remove(VanillaEffects::HEALTH_BOOST());
        $this->sendActivation($attacker, "§cStruck by §bThunderbolt §cEnchant! Health Effects removed");
        $this->sendActivation($player, "§bThunderbolt §aActivated!");
        $this->pl->strikeLightning($attacker, 5);
        foreach ($player->getWorld()->getNearbyEntities($player->getBoundingBox()->expandedCopy(5, 5, 5)) as $p) {
            if ($p instanceof Player) {
                $this->pl->strikeLightning($p, 1);
            }
        }
    }

}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/sword/Thunderbolt.java
package dev.skyblock.enchants.sword;
import dev.skyblock.enchants.BaseMeleeEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
public class Thunderbolt extends BaseMeleeEnchant {
    public static final int ID = 179;
    public Thunderbolt() { super(ID, "Thunderbolt", 15); }
    @Override
    public boolean isApplicableTo(Player holder, int level) { return java.util.concurrent.ThreadLocalRandom.current().nextInt(1, 26) == 1; }
    @Override
    public void onActivation(Player victim, Player attacker, EntityDamageByEntityEvent ev, int level) {
        attacker.sendMessage("§bThunderbolt §aActivated!");
        victim.getWorld().strikeLightning(victim.getLocation());
        ev.setDamage(ev.getDamage() + level);
    }
}
